test/Vault.spec.ts

```diff
@@ -322,6 +325,41 @@ describe('Vault', () => {
     });
   });
 
+  describe('transferOwnership', () => {
+    it('can only be called by the current owner', async () => {
+      await expect(
+        vault.connect(alice).transferOwnership(alice.address),
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+
+    it('reverts if new owner is address(0)', async () => {
+      await expect(
+        vault.connect(owner).transferOwnership(constants.AddressZero),
+      ).to.be.revertedWith('VaultOwnerCannotBe0Address');
+    });
+
+    it('changes ownership to the new owner', async () => {
+      await vault.connect(owner).transferOwnership(alice.address);
+
+      expect(await vault.owner()).to.be.equal(alice.address);
+    });
+
+    it("revokes all previous owner's roles and sets them for the new owner", async () => {
```

Comment:
Not sure about that one since the only place these roles are assigned is in the constructor and we already have tests for that. I can do the change if you think it is necessary

Reply:
i think it's good practice because it makes the tests depend on each other less and when a change is made to the constructor for example having these assertions will tell you more clearly where the problem is

Reply:
yeah, this is a bit tricky, but I can imagine ourselves messing this up

Reply:
Hmm, I get your point, maybe I'm missing something but I think a little bit differently. In the example you described, we will have more dependency between tests. Let's say that role assignments in the constructor change, this will cause the test for transferOwnershiip to fail, even though nothing in the transferOwnersip code has changed.
Interesting discussion btw :slightly_smiling_face: 
@gabrielpoca @dxganta what do you guys think about this?

Reply:
but that's exactly what we want, we want this test to fail so you get an opportunity to revisit this functionality and ensure it's still correct